benchmarks/dynamo/common.py

```diff
@@ -242,6 +246,16 @@ def timed(model, model_iter_fn, example_inputs, times=1, return_result=False):
     # Dont collect outputs to correctly measure timing
     for _ in range(times):
         result = model_iter_fn(model, example_inputs, collect_outputs=False)
+        if tensor_is_on_xla(result):
+            # If the model is on XLA device, it's possible that after running
+            # the model, the computation is accumulated but not performed yet.
```

Comment:
:D if model is on XLA_DEVICE, it is almost guarantee that computation won't be executed by just running the model(unless all ops fallback), I guess we can make a stronger statement here.

Reply:
hmm, but what if we run a cached graph? In that case, the result will be DeviceData nodes and the computation is already performed?

Reply:
cached graph is a bit ambigous. If it is a cache graph in lazy concept, we still do the tracing of the IR graph, but only skip the `IR->HLO->executable`. In this case, without a `mark_step`, result will still be a pending IR graph.

If we are talking about cached graph in dynamo concept, we actually skip all tracing and execute the graph directly. I thought the comment "running the model" means under lazy concept.

Reply:
This method is used for both baseline and optimized code path to measure execution time. So we should expect both scenarios you mentioned

Reply:
sg!